lib/models/repository-states/present.js

```diff
@@ -45,11 +45,7 @@ export default class Present extends State {
   }
 
   setCommitMessage(message) {
-    const oldMessage = this.commitMessage;
     this.commitMessage = message;
-    if (oldMessage !== message) {
-      this.didUpdate();
-    }
```

Comment:
Oh, interesting. Changing the message triggered an update, and therefore at least a partial re-render of the React tree, so that UI elements that change state based on the message's length and presence have a chance to be updated. Specifically I'm thinking of:

* The enablement of the "Commit" button; it should be disabled if the message is empty; and
* The "remaining characters" counter that changes as you type the subject line;

But, the way the commit message is handled has been shuffled a lot with the introduction of the `AtomTextEditor` wrapper component and various refactorings, so I'm not 100% sure that these things _won't_ be updated by something else.

Can you confirm that those didn't break? If they still work, then we definitely should take this out because it's not needed. If they don't, an alternate approach would be to fuss with either the `UserStore` or the `GitTabController` to fine-tune when `loadUsersFromLocalRepo` is invoked. We should only really need to re-call it when something has changed that would change the available co-authors, like a fetch or a commit.

Reply:
Looking a little more closely at @maxbrunsfeld's flame graph, one thing that stands out to me is that we're calling `setState()` and running an entire render cycle every time `loadUsersFromLocalRepo` finishes, even though the result is the same. I'm guessing that's the `setState` call in here:

https://github.com/atom/github/blob/df5ef8c48f0fe833772a6212200a5490c09bcdfa/lib/controllers/git-tab-controller.js#L69-L74

Maybe we should also prevent `UserStore.onDidUpdate()` from firing when the user list hasn't actually changed... ?

Reply:
@smashwilson yes, my original solution was for the `UserStore` to not invoke `loadUsersFromLocalRepo` when not needed.

However, I noticed we were also performing a bunch of other unnecessary operations on the repository when the commit message was updated (, and figured it would be a better performance improvement to fix the problem here instead of in the user store.

I did some additional testing and confirmed that the commit button is still disabled if the message is empty, and the remaining characters counter changes as you type in the commit box.  Which makes sense, because the `CommitView` component has its own logic for dealing with commit message updates, that forces a re render when the message has changed.

Reply:
> Which makes sense, because the CommitView component has its own logic for dealing with commit message updates, that forces a re render when the message has changed.

:sparkles: This was the piece that I wasn't sure was in place or not. Excellent :cake: